libs/logging/src/types.ts

```diff
@@ -41,5 +46,103 @@ export interface LogLine extends Dictionary<string> {
   user: LoggingUserRole;
   disposition: LogDisposition;
   message?: string;
-  time?: string;
+  timeLogInitiated?: string;
+}
+
+export const LogSourceSchema: z.ZodSchema<LogSource> = z.nativeEnum(LogSource);
+export const LogEventIdSchema: z.ZodSchema<LogEventId> =
+  z.nativeEnum(LogEventId);
+export const LogEventTypeSchema: z.ZodSchema<LogEventType> =
+  z.nativeEnum(LogEventType);
+export const LoggingUserRoleSchema: z.ZodSchema<LoggingUserRole> = z.union([
+  CardDataTypesSchema,
+  z.literal('vx-staff'),
+  z.literal('system'),
+  z.literal('unknown'),
+]);
+
+export const LogLineSchema: z.ZodSchema<LogLine> = z.object({
+  source: LogSourceSchema,
+  eventId: LogEventIdSchema,
+  eventType: LogEventTypeSchema,
+  user: LoggingUserRoleSchema,
+  disposition: z.string(),
+  message: z.string().optional(),
+  timeLogInitiated: z.string().optional(),
+  // this key is not required when writing logs but we want to make sure it is present when reading logs
+  timeLogWritten: z.string(),
+});
+
+// Enumeration of valid DeviceType's from the NIST Election Event Logging CDF Specification
+export enum CdfDeviceType {
+  Adjudication = 'adjudication',
+  BallotActivation = 'ballot-activation',
+  Bmd = 'bmd',
+  BallotPrinting = 'ballot-printing',
+  BlankBallotPrinting = 'blank-ballot-printing',
+  Dre = 'dre',
+  DreController = 'dre-controller',
+  ElectronicCast = 'electronic-cast',
+  ElectronicCastPaper = 'electronic-cast-paper',
+  ElectronicCastBook = 'electronic-cast-book',
+  Ems = 'ems',
+  ScanBatch = 'scan-batch',
+  ScanSingle = 'scan-single',
+  TransmissionSending = 'transmission-sending',
+  TransmissionReceiving = 'transmission-receiving',
+  Other = 'other',
+}
+
+export const DEVICE_TYPES_FOR_APP: Dictionary<CdfDeviceType> = {
+  [LogSource.VxAdminApp]: CdfDeviceType.Ems,
+  [LogSource.VxBatchScanApp]: CdfDeviceType.ScanBatch,
+  [LogSource.VxPrecinctScanApp]: CdfDeviceType.ScanSingle,
+  [LogSource.VxBallotMarkingDeviceApp]: CdfDeviceType.Bmd,
+  [LogSource.VxBallotActivationApp]: CdfDeviceType.BallotActivation,
+};
+
+export enum CdfHashType {
+  Md6 = 'md6',
+  Sha256 = 'sha-256',
+  Sha512 = 'sha-512',
+  Other = 'other',
+}
+
+export enum CdfEventDispositionType {
+  Failure = 'failure',
+  Na = 'na',
+  Success = 'success',
+  Other = 'other',
+}
+
+export interface CdfEvent {
+  id: LogEventId;
+  disposition: CdfEventDispositionType;
+  sequence: string;
+  timestamp: string;
+  type: LogEventType;
+  details?: string;
+  description?: string;
+  hash?: string;
+  otherDisposition?: string;
+  severity?: string;
+  userId: string;
```

Comment:
These are `lowerCamelCase` whereas the [CDF JSON schema](https://github.com/usnistgov/ElectionEventLogging/blob/master/NIST_V1_election_event_logging.json) uses `UpperCamelCase`. So these won’t actually be compatible with something expecting CDF logging data, right?

Reply:
Yeah will fix, I got thrown off by this not matching the style-guide rules. I wish I had an example CDF file :'( 

Do you think from that that things like "Device" need to be prefixed with "ElectionLogging." as well?